dev/DynamicDependency/API/MsixDynamicDependency.h

```diff
@@ -8,10 +8,18 @@
 
 #include <stdint.h>
 
-// MSIX Dynamic Dependency HRESULT: Data Store not found (Windows App Runtime's Main package not registered?)
+/// MSIX Dynamic Dependency HRESULT: Windows App Runtime is not in the package graph.
+#define MDD_E_WINDOWSAPPRUNTIME_NOT_IN_PACKAGE_GRAPH    _HRESULT_TYPEDEF_(0x80040001L)
```

Comment:
I understand we should maintain back compat, but reusing/creating HRESULTs is not a good pattern.  Running err in razzle gives this output: # for hex 0x80040001 / decimal -2147221503
  E_DISCVOL_VOLUME_TOO_SMALL                                     discvol.h
  DU_E_OUTOFKERNELRESOURCES                                      dusererror.h
# There are not enough Kernel resources to perform the
# operation
  E_COLORCPL_NO_SUCH_DEVICE                                      icmpriv.h
  E_NOTVALIDFORANIMATEDIMAGE                                     shimgdata.h
  TAPI_E_NOTENOUGHMEMORY                                         tapi3err.h
# The buffer passed in to this method was not big enough.
  IAP_E_BAD_PRODUCTID_LIST                                       windowsstoresdkerrors.h
# The supplied product list is bad. Check the Store event log
# for more details.
  OLE_E_ADVF                                                     winerror.h
# Invalid advise flags
  XAUDIOPACKETSTATUS_XMA_BUFFER_ERROR                            xauddefs.h
# as an HRESULT: Severity: FAILURE (1), FACILITY_IO_ERROR_CODE (0x4), Code 0x1
# for hex 0x1 / decimal 1
  STATUS_WAIT_1                                                  ntstatus.h
# as an HRESULT: Severity: FAILURE (1), FACILITY_ITF (0x4), Code 0x1
  ERROR_INVALID_FUNCTION                                         winerror.h
# Incorrect function.
# 10 matches found for "80040001"

Reply:
> I understand we should maintain back compat, but reusing/creating HRESULTs is not a good pattern.  

Yes and no. 

Yes, it's good. This HRESULT is a FACILITY_ITF (4) which is an open namespace. ALL 0x8004xxxx are, by definition, not guaranteed unique across space and time and dependent on current context. This is how you define new HRESULTs if you're not inside the Windows build and can stake out new unique values in winerror.h

That said, No it's bad as guidelines recommend only using values 0x0200 - 0xFFFF

https://learn.microsoft.com/en-us/windows/win32/com/codes-in-facility-itf

> All the COM-defined FACILITY_ITF codes have a code value in the range of 0x0000-0x01FF. While it is legal to use any codes in FACILITY_ITF, it is recommended that only code values in the range of 0x0200-0xFFFF be used. This recommendation is made as a means of reducing confusion with any COM-defined errors.

I'd like to change the value but was waiting for 2.0 when breaking changes are acceptable.

Unless you feel strongly about it and approve breaking the usual compat policy in minor updates? If so I'll make the change in main for 1.6

Reply:
I like some of the contrary responses here: https://stackoverflow.com/questions/53199912/encoding-hresult-values-use-facility-itf-or-set-bit-29

I.e., mostly "no" 

I agree if we're going to change HRESULTs, it would have to wait until 2.0.  I'd suggest we shouldn't even do it then.  Ideally, we would reuse well-known HRESULTs that have some semblance of the WAR operation failure.  But we are stuck with these.